File: Job-Portal-BackEnd/src/main/java/lk/ijse/jobportal/service/ApplyJobService.java
package lk.ijse.jobportal.service;

import lk.ijse.jobportal.dto.ApplyJobDTO;
import lk.ijse.jobportal.dto.ApplyJobMainDTO;
import org.springframework.http.ResponseEntity;

public interface ApplyJobService {

    public ResponseEntity<?> applyJob(ApplyJobMainDTO applyJobMainDTO);

    public ResponseEntity<?> getAllAppliedJobsByUser(String username);

    public ResponseEntity<?> getAllAppliedEmployeeByUser(String username);

    public ApplyJobDTO getLastData();
}

File: Job-Portal-FrontEnd/src/app/service/apply-job.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from "@angular/common/http";
import {Observable} from "rxjs/index";
import {ApplyJobMain} from "../dto/apply-job-main";
import {MAIN_URL} from "./user-service.service";
import {ApllyJob} from "../dto/aplly-job";

const URLS="/api/v1/applyJob"
const URLFile="/file"
@Injectable()
export class ApplyJobService {

  constructor(private http:HttpClient) { }

  saveApplyJob(applyJob:ApplyJobMain):Observable<any>{
    return this.http.post<any>(MAIN_URL+URLS,applyJob);
  }

  getAllAppliedJobsByUser(user:String){
    return this.http.get<any>(MAIN_URL+URLS+"/getAllAppliedJobsByUser/"+user)
  }

  getAllAppliedEmployeeByUser(user:String){
    return this.http.get<any>(MAIN_URL+URLS+"/getAllAppliedEmployeeByUser/"+user)
  }

  getFile(path:String){
    const formData:FormData=new FormData();
    // @ts-ignore
    formData.append("path",path);
    return this.http.post<any>(MAIN_URL+URLFile,formData);
  }

  getLastJob():Observable<ApllyJob>{
    return this.http.get<ApllyJob>(MAIN_URL+URLS+"/Last")
  }
}
